dmutils/email/dm_notify.py

```diff
@@ -17,21 +17,35 @@ class DMNotifyClient:
 
     def __init__(
             self,
-            govuk_notify_api_key,
+            govuk_notify_api_key=None,
             govuk_notify_base_url='https://api.notifications.service.gov.uk',
-            logger=None,
             redirect_domains_to_address=None,
+            *,
+            logger=None,
+            templates=None,
     ):
         """
-            :param govuk_notify_api_key:
+            :param govuk_notify_api_key: defaults to current_app.config["DM_NOTIFY_API_KEY"]
             :param govuk_notify_base_url:
-            :param logger: logger to log progress to, taken from current_app if Falsey
             :param redirect_domains_to_address: dictionary mapping email domain to redirected email address - emails
                 sent to a email with a domain in this mapping will instead be sent to the corresponding value set here.
                 if `redirect_domains_to_address` is `None` will fall back to looking for a
                 `DM_NOTIFY_REDIRECT_DOMAINS_TO_ADDRESS` setting in the current flask app's config (if available).
+
+        The following arguments are keyword-only and should only be used if you want to operate outside
+        of a Flask app context.
+            :param logger: logger to log progress to, taken from current_app if Falsey
+            :param templates: a dictionary of template names to template uuids, so that you can use
+                descriptive names when specifying a template. This defaults to current_app.config["NOTIFY_TEMPLATES"].
         """
+        if govuk_notify_api_key is None:
+            govuk_notify_api_key = current_app.config["DM_NOTIFY_API_KEY"]
```

Comment:
If we're doing this, we can also refactor `dmutils.email.user_account_email.send_user_account_email` to remove the key from the client initialisation. That function also is specifying `template_id` as a kwarg on `send_email`...

Reply:
Yeah I totally missed that 😳 There is a PR I'm working on now that will fix that bug (#456)